docs/DeveloperGuide.md

```diff
@@ -7,26 +7,42 @@ title: Developer Guide
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Acknowledgements**
+## **Acknowledgements** (OUTDATED)
 
 * {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Setting up, getting started**
+## **Setting up, getting started** (OUTDATED)
 
 Refer to the guide [_Setting up and getting started_](SettingUp.md).
 
 --------------------------------------------------------------------------------------------------------------------
 
 ## **Design**
 
+RC4HDB has the idea of providing a flexible product which functions well, yet not too complex. Keeping this in mind, 
+the developers are pursuing an iterative approach, adding new features and functionalities amidst the evolving requirements.
+This gives rise to the following main guiding principles for RC4HDB: 
+
+**Maintainability**
+
+This project was adapted from an application called [`AddressBook Level 3 (AB3)`](https://se-education.org/addressbook-level3/).
+The developers of `AB 3` have developed in a manner that facilitates easy modification of components. This design allows the functionalities
+implemented to be easily changed depending on the goals of the developers. Building upon the existing components in `AB3`, the developers are to
+add additional classes to the major components including [**`UI`**](#ui-component), [**`Logic`**](#logic-component) , [**`Model`**](#model-component), [**`Storage`**](#storage-component).
+
+**Command Line Interface (CLI) Oriented**
+
+[**CLI**](#glossary) gives the user an easy way to type commands. This is especially useful for a target audience which is familiar with the process of performing admin tasks
+using a computer. For users who type fast, RC4HDB will be highly efficient and quick to respond, replacing their existing processes of managing their housing database.
```

Comment:
Maybe "**... replacing their existing processes of managing their housing database.**" might be too strong of a statement. Perhaps something like "**satisfying the user's housing management database needs.**"

Once again, this is really subjective, so use your judgement on this, I'm fine with either.

Reply:
I decided to change it to "**improving their existing processes of managing their housing database.**"